okhttp/src/main/java/okhttp3/internal/http/RealInterceptorChain.kt

```diff
@@ -84,7 +84,7 @@ class RealInterceptorChain(
     calls++
 
     if (exchange != null) {
-      check(exchange.connection.supportsUrl(request.url)) {
+      check(exchange.finder.sameHostAndPort(request.url)) {
```

Comment:
We could technically get by without this one, the network interceptor that is changing the request may have prior knowledge that the request url change is within the same server?

Reply:
Yep. But I think it’s safer to just forbid it and wait for that request to come to us. Otherwise we’re also bypassing certificate pinning on b.example.com and that’s awkward.